### DISCUSSION TITLE
start sftp service in pod with s3 storage and ldap auth

### DISCUSSION THREAD

**Discussion by tooptoop4** [drakkan/sftpgo#819] [General]
this tool seems to do a lot so I wasn't sure if it would meet my use case. similar to https://github.com/Turgon37/docker-sftp-ldap I want to launch a pod that exposes sftp daemon so remote hosts can sftp files to this pod (authenticating via ldap user/pw) but have the files stored on s3 instead of local pod storage

**Comment by drakkan** [comment]
Hi,

to authenticate against LDAP you need a custom authentication [hook](https://github.com/drakkan/sftpgo/blob/main/docs/external-auth.md) or plugin. Some examples are available:

https://github.com/drakkan/sftpgo/blob/main/examples/ldapauth
https://github.com/drakkan/sftpgo/blob/main/examples/ldapauthserver
https://github.com/orware/sftpgo-ldap-http-server
https://www.youtube.com/watch?v=M5UcJI8t4AI

No one has been interested enough to sponsor an "official" plugin until now

**Reply by tooptoop4** [reply]
@drakkan are the gid memberships inherited by ldap users authenticating?

**Reply by drakkan** [reply]
> @drakkan are the gid memberships inherited by ldap users authenticating?

You have to implement your own hook/plugin so you can do what you prefer with group membership, for example @orware added some folders based on group membership, see [here](https://github.com/orware/sftpgo-ldap-http-server/blob/main/configuration.example.php#L180)
